pyquil/operator_estimation.py

```diff
@@ -768,27 +798,48 @@ def measure_observables(qc: QuantumComputer, tomo_experiment: TomographyExperime
         for qubit, op_str in max_weight_out_op:
             total_prog += _local_pauli_eig_meas(op_str, qubit)
 
-        # 2. Run the experiment
-        bitstrings = qc.run_and_measure(total_prog, n_shots)
+        if symmetrize:
+            # 1.4 Symmetrize -- flip qubits pre-measurement
+            qubits = max_weight_out_op.get_qubits()
+            ops_strings = _ops_strs_symmetrize(qubits)
+            list_bitstrings_symm = []
+            for ops_str in ops_strings:
+                total_prog_symm = total_prog.copy()
+                prog_symm = _ops_str_to_prog(ops_str, qubits)
+                total_prog_symm += prog_symm
+                # 2. Run the experiment
+                n_shots_symm = n_shots // len(ops_strings)
+                bitstrings_symm = qc.run_and_measure(total_prog_symm, n_shots_symm)
+                # 2.1 Flip the results post-measurement
+                d_flips_symm = _ops_str_to_flips(ops_str, qubits)
+                for qubit, bs_results in bitstrings_symm.items():
```

Comment:
this is like a 3- or 4- nested python for loop over potentially thousands of items. I suspect this could cause a noticeable performance penalty. Consider using vectorized numpy operations to do the flip-back

Reply:
This loop is O(number of qubits), not thousands (*). It's necessitated by the dict structure of `qc.run_and_measure(..)`. It would be nice to rewrite the entire thing to use `qc.run(..)` with explicit `MEASURE` instructions instead, but I'll leave that for a future PR.

(*) The outer loop over settings could still be in the thousands for some situations e.g. when the number of these scales as 3^n, so it would be useful to test performance in such situations.